app/Http/Controllers/Admin/ModulesController.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+namespace App\Http\Controllers\Admin;
+
+use App\Contracts\Controller;
+use App\Models\Module;
+use App\Services\ModuleService;
+use Illuminate\Http\Request;
+use Laracasts\Flash\Flash;
+
+class ModulesController extends Controller
+{
+    private $moduleSvc;
+
+    public function __construct(ModuleService $moduleSvc)
+    {
+        $this->moduleSvc = $moduleSvc;
+    }
+
+    /**
+     * Display a listing of the resource.
+     *
+     * @return mixed
+     */
+    public function index()
+    {
+        $modules = Module::all()->sortByDesc('id');
+        $new_modules = $this->moduleSvc->scan();
+        return view('admin.modules.index', [
+            'modules'     => $modules,
+            'new_modules' => $new_modules,
+        ]);
+    }
+
+    public function create()
+    {
+        return view('admin.modules.create');
+    }
+
+    public function store(Request $request)
+    {
+        $array = [
+            'file'    => $request->file('module_file'),
+            'enabled' => $request->has('enabled'),
```

Comment:
I don't think you need to pass 'enabled' here. So you can just do `$this->moduleSvc->installModule($request->file('module_file');`. We can simplify this - by uploading a file, the implication is that it will be enabled by default

Reply:
Okay. I will remove option to enable/disable module from admin panel then.